File: tests/fixtures/sample.go
// Sample Go file for testing
package main

import "fmt"

type User struct {
    ID    int
    Name  string
    Email string
}

type UserService struct {
    users map[int]User
}

func NewUserService() *UserService {
    return &UserService{
        users: make(map[int]User),
    }
}

func (s *UserService) AddUser(user User) {
    s.users[user.ID] = user
}

func (s *UserService) GetUser(id int) (User, bool) {
    user, ok := s.users[id]
    return user, ok
}

func main() {
    service := NewUserService()
    service.AddUser(User{ID: 1, Name: "Alice", Email: "alice@example.com"})
    user, ok := service.GetUser(1)
    if ok {
        fmt.Printf("Found user: %s\n", user.Name)
    }
}

File: tests/fixtures/sample.rs
// Sample Rust file for testing
use std::collections::HashMap;

struct User {
    id: u32,
    name: String,
    email: String,
}

struct UserService {
    users: HashMap<u32, User>,
}

impl UserService {
    fn new() -> Self {
        UserService {
            users: HashMap::new(),
        }
    }

    fn add_user(&mut self, user: User) {
        self.users.insert(user.id, user);
    }

    fn get_user(&self, id: u32) -> Option<&User> {
        self.users.get(&id)
    }

    fn for_each<F>(&self, mut callback: F)
    where
        F: FnMut(&User),
    {
        for user in self.users.values() {
            callback(user);
        }
    }
}

fn main() {
    let mut service = UserService::new();
    service.add_user(User {
        id: 1,
        name: String::from("Alice"),
        email: String::from("alice@example.com"),
    });
    if let Some(user) = service.get_user(1) {
        println!("Found user: {}", user.name);
    }
}
